app/models/spree_avatax_certified/avatax_log.rb

```diff
@@ -1,59 +1,40 @@
 module SpreeAvataxCertified
   class AvataxLog
+    attr_reader :logger
+
     def initialize(file_name, log_info = nil, schedule = nil)
-      if !Spree::Config.avatax_log_to_stdout
-        schedule = 'weekly' unless schedule != nil
-        @logger ||= Logger.new("#{Rails.root}/log/avatax.log", schedule)
-        progname(file_name.split('/').last.chomp('.rb'))
-        info(log_info) unless log_info.nil?
-      else
-        log_info = "-#{file_name} #{log_info}"
+      unless Spree::Config.avatax_log_to_stdout
         @logger ||= Logger.new(STDOUT)
+        return
       end
-    end
 
-    def logger
-      @logger
+      @logger ||= Logger.new(Rails.root.join('log', 'avatax.log'),
+                             schedule.presence || 'weekly')
+
+      progname(file_name.split('/').last.chomp('.rb'))
+      info(log_info) if log_info
     end
 
     def enabled?
       Spree::Config.avatax_log || Spree::Config.avatax_log_to_stdout
     end
 
     def progname(progname = nil)
-      if enabled?
-        progname.nil? ? logger.progname : logger.progname = progname
-      end
-    end
+      return unless enabled?
 
-    def info(message, obj = nil)
-      if enabled?
-        logger.info "[AVATAX] #{message} #{obj}"
-      end
+      progname.nil? ? logger.progname : logger.progname = progname
```

Comment:
Please rewrite this to regular if/else statement and rebase against current master